build-info-extractor-gradle/src/test/java/org/jfrog/gradle/plugin/artifactory/Utils.java

```diff
@@ -89,8 +89,20 @@ static BuildResult runGradle(String gradleVersion, Map<String, String> envVars,
      * @param publishBuildInfo - Publish build info
      * @throws IOException - In case of any IO error
      */
-    static void generateBuildInfoProperties(String contextUrl, String username, String password, String localRepo, String virtualRepo, String publications, boolean publishBuildInfo) throws IOException {
-        String content = new String(Files.readAllBytes(BUILD_INFO_PROPERTIES_SOURCE), StandardCharsets.UTF_8);
+    static void generateBuildInfoProperties(String contextUrl, String username, String password, String localRepo, String virtualRepo, String publications, boolean publishBuildInfo, boolean setDeployer, boolean setResolver) throws IOException {
+        String content = "";
+        if (setDeployer) {
+            content += generateBuildInfoPropertiesForServer(contextUrl, username, password, localRepo, virtualRepo, publications, publishBuildInfo, BUILD_INFO_PROPERTIES_SOURCE_DEPLOYER);
+            content += "\n";
+        }
+        if (setResolver) {
+            content += generateBuildInfoPropertiesForServer(contextUrl, username, password, localRepo, virtualRepo, publications, publishBuildInfo, BUILD_INFO_PROPERTIES_SOURCE_RESOLVER);
+        }
+        Files.write(BUILD_INFO_PROPERTIES_TARGET, content.getBytes(StandardCharsets.UTF_8));
+    }
+
+    static String generateBuildInfoPropertiesForServer(String contextUrl, String username, String password, String localRepo, String virtualRepo, String publications, boolean publishBuildInfo, Path source) throws IOException {
```

Comment:
Let's add a description for this method too. It stands out, because its neighbouring methods have complete descriptions.
Also, it looks like this method can be set as private.